### DISCUSSION TITLE
VipsForeignLoad: "bmptest.bmp" is not a known file format

### DISCUSSION THREAD

**Discussion by JeanPenty** [libvips/libvips#3994] [General]
use version 8.15.2

			VipsImage* image = vips_image_new();
			image = vips_image_new_from_file("bmptest.bmp");
			if (image)
			{
				int left = 100;  // clip left
				int top = 100;    // clip top
				int width = image->Xsize - 200;  // clip width
				int height = image->Ysize - 200; // clip height
				VipsImage* cropped = vips_image_new();
				vips_crop(image, &cropped, left, top, width, height, NULL);
				vips_image_write_to_file(cropped, "cropped.png", NULL);
			}
			else
			{
				//
				std::ostringstream os;
				os << vips_error_buffer();
				std::string strErr = os.str();
			}

load bmp file failed
error：VipsForeignLoad: "bmptest.bmp" is not a known file format

**Comment by lovell** [comment]
BMP bitmap images are decoded via ImageMagick and you'll need to use a version of libvips that has been compiled with such support.

You can see which formats/decoders are available by running `vips --vips-config` at the command line.

**Reply by JeanPenty** [reply]
OK, thank you so much
